test/test_jit_cuda_fuser.py

```diff
@@ -4233,6 +4233,24 @@ def t(x_kj: torch.Tensor, idx_kj: torch.Tensor, sbf: torch.Tensor, dim : int):
         t_jit = torch.jit.script(t)
         self._run_helper(t_jit, t, lookup_tv, indies_tv, sbf, dim)
 
+    @unittest.skipIf(not RUN_NVFUSER, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING,
+                     "Requires fusion optimization pass to be effective")
+    def test_gather_fusion(self):
+        lookup_size = 68
+        feat_dim = 128
+        select_dim = 1
+        lookup_tv = torch.rand(lookup_size, feat_dim, dtype=torch.float, device="cuda")
+        indies_tv = torch.randint(0, lookup_size, (lookup_size, select_dim), device="cuda").to(dtype=torch.long)
+        sbf = torch.rand(lookup_size, select_dim, dtype=torch.float, device="cuda")
+
+        def t(x_kj, idx_kj, inp):
+            gather_res = torch.gather(x_kj, 1, idx_kj) * inp
+            res = gather_res + 17
+            return res
+        t_jit = torch.jit.script(t)
+        self._run_helper(t_jit, t, lookup_tv, indies_tv, sbf)
```

Comment:
We are not testing autodiff here. Can we try with `_run_training_helper`?

I would also suggest us testing `sparse_grad` here as well (with it being false & true).

Reply:
Done in new commit.